From 72c6cc88ee2eb48a9d99203b440b66562c9ba45c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Fri, 21 Aug 2026 20:00:58 +0200
Subject: [PATCH] systray: Avoid crash when bus_name contains garbage

We should return TRUE in this case as well, as we actually handle the
invocation, although by returning an error. Otherwise it makes glib
crash.

Related: #992
(cherry picked from commit 1c9220f8eeb7aa093a9b107116705ed2ac3c28d4)
---
 plugins/systray/sn-backend.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/systray/sn-backend.c b/plugins/systray/sn-backend.c
index fd8c851af..0a3b4a817 100644
--- a/plugins/systray/sn-backend.c
+++ b/plugins/systray/sn-backend.c
@@ -375,7 +375,7 @@ sn_backend_watcher_register_item (SnWatcher *watcher_skeleton,
                                                      G_IO_ERROR,
                                                      G_IO_ERROR_INVALID_ARGUMENT,
                                                      "Invalid bus name");
-      return FALSE;
+      return TRUE;
     }
 
   key = g_strdup_printf ("%s%s", bus_name, object_path);
-- 
GitLab

